src/preferences/dialog/dlgpreferences.cpp

```diff
@@ -419,15 +418,16 @@ void DlgPreferences::onShow() {
     int newX = m_geometry[0].toInt();
     int newY = m_geometry[1].toInt();
 
-    const QScreen* primaryScreen = getPrimaryScreen();
+    const QScreen* const pScreen = mixxx::widgethelper::getScreen(*this);
     QSize screenSpace;
-    if (primaryScreen) {
-        screenSpace = primaryScreen->geometry().size();
-    } else {
+    VERIFY_OR_DEBUG_ASSERT(pScreen) {
         qWarning() << "Assuming screen size of 800x600px.";
         screenSpace = QSize(800, 600);
     }
-    newX = std::max(0, std::min(newX, screenSpace.width()- m_geometry[2].toInt()));
+    else {
+        screenSpace = pScreen->availableSize();
```

Comment:
I can confirm that both classes are returning the same values for size and available size. 

Here you have replaced:
primaryScreen->geometry().size();
with:
screenSpace = pScreen->availableSize();

Please revert, If you are not interested in changing the math.

Reply:
> windowMargin(Horz/Vert)/(X/Y)/(Width/Height)?

windowMarginWidth, windowMarginHeight works for me.